### PR TITLE
component solder ball getter and setter added

### LINKED ISSUE #238: edb.components.set_solder_ball shape "None" not available through EDB API
### Description of the feature

It would be helpful if all available options are presented to EDB API. "None" is not available for solder ball shapes but this can be done manually in HFSS 3D Layout.

### Steps for implementing the feature

Add "None" to edb.components.set_solder_ball shape types.

### Useful links and references

This can be selected manually in HFSS 3D Layout

image

### PR DESCRIPTION
**Pull request by svandenb-dev**
getter/setter added:
- solder ball shape
- solder ball height
- solder ball diameter

Close #238

### CODE REVIEW

File: src/pyedb/dotnet/edb_core/edb_data/components_data.py [hunk 1]
```diff
@@ -297,6 +302,82 @@ def solder_ball_height(self):
...
+                shape = self._edb.definition.SolderballShape.NoSolderball
+            elif value.lower() == "spheroid":
+                shape = self._edb.definition.SolderballShape.Spheroid
+        if isinstance(value, int):
+            if value == 0:
+                shape = self._edb.definition.SolderballShape.NoSolderball
+            if value == 1:
+                shape = self._edb.definition.SolderballShape.Cylinder
+            if value == 2:
+                shape = self._edb.definition.SolderballShape.Spheroid
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/components_data.py:344]:
```suggestion
            elif value == 1:
                shape = self._edb.definition.SolderballShape.Cylinder
            elif value == 2:
                shape = self._edb.definition.SolderballShape.Spheroid
```

File: src/pyedb/dotnet/edb_core/edb_data/components_data.py [hunk 2]
```diff
@@ -297,6 +302,82 @@ def solder_ball_height(self):
...
+    @solder_ball_height.setter
+    def solder_ball_height(self, value):
+        if "GetSolderBallProperty" in dir(self.component_property):
+            if value:
+                sball_height = round(self._edb.utility.Value(value).ToDouble(), 9)
+                cmp_property = self.component_property
+                solder_ball_prop = cmp_property.GetSolderBallProperty().Clone()
+                solder_ball_prop.SetHeight(self._get_edb_value(sball_height))
+                cmp_property.SetSolderBallProperty(solder_ball_prop)
+                self.component_property = cmp_property
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/components_data.py:314]:
```suggestion
        if "GetSolderBallProperty" in dir(self.component_property) and value:
            sball_height = round(self._edb.utility.Value(value).ToDouble(), 9)
            cmp_property = self.component_property
            solder_ball_prop = cmp_property.GetSolderBallProperty().Clone()
            solder_ball_prop.SetHeight(self._get_edb_value(sball_height))
            cmp_property.SetSolderBallProperty(solder_ball_prop)
            self.component_property = cmp_property
```

File: src/pyedb/dotnet/edb_core/edb_data/components_data.py [hunk 3]
```diff
@@ -297,6 +302,82 @@ def solder_ball_height(self):
...
+            cmp_property.SetSolderBallProperty(solder_ball_prop)
+            self.component_property = cmp_property
+
+    @property
+    def solder_ball_diameter(self):
+        """Solder ball diameter"""
+        if "GetSolderBallProperty" in dir(self.component_property):
+            result = self.component_property.GetSolderBallProperty().GetDiameter()
+            if result[0]:
+                return result[1], result[2]
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/components_data.py:358]: This is hardly readable. I assume that result[0] is a boolean but what are result[1] and [2] ? Are they "end" and "mid" diameters ?

File: src/pyedb/dotnet/edb_core/edb_data/components_data.py [hunk 4]
```diff
@@ -297,6 +302,82 @@ def solder_ball_height(self):
...
+            if len(value) == 2:
+                val1 = self._get_edb_value(value[0])
+                val2 = self._get_edb_value(value[1])
+            elif len(value) == 1:
+                val1 = self._get_edb_value(value[0])
+                val2 = self._get_edb_value(value[0])
+        if isinstance(value, str):
+            val1 = self._get_edb_value(value)
+            val2 = self._get_edb_value(value)
+        if val1 and val2:
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/components_data.py:374]:
```suggestion
        elif isinstance(value, str):
            val1 = self._get_edb_value(value)
            val2 = self._get_edb_value(value)
        elif val1 and val2:
```

File: src/pyedb/dotnet/edb_core/edb_data/components_data.py [hunk 5]
```diff
@@ -297,6 +302,82 @@ def solder_ball_height(self):
...
+        """Solder ball diameter"""
+        if "GetSolderBallProperty" in dir(self.component_property):
+            result = self.component_property.GetSolderBallProperty().GetDiameter()
+            if result[0]:
+                return result[1], result[2]
+
+    @solder_ball_diameter.setter
+    def solder_ball_diameter(self, value):
+        val1 = None
+        val2 = None
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/edb_data/components_data.py:363]: Same comment as previous one

**Review by SMoraisAnsys — CHANGES_REQUESTED** [review]

File: src/pyedb/dotnet/edb_core/edb_data/components_data.py (hunk 3 above)
Comment (svandenb-dev) [reply src/pyedb/dotnet/edb_core/edb_data/components_data.py:358]: Yes this is the output from internal EDB api

Comment (svandenb-dev) [reply src/pyedb/dotnet/edb_core/edb_data/components_data.py:358]: result[1] and [2] are 2 diameters (main and mid one) 
image
Using overload also

**Review by SMoraisAnsys — DISMISSED** [review]
LGTM, good work

**Review by SMoraisAnsys — APPROVED** [review]

**Review by SMoraisAnsys — CHANGES_REQUESTED** [review]
Good job here !

**Review by SMoraisAnsys — APPROVED** [review]
LGTM

**Pull request merged**